### PR TITLE
Move chalang-related docs into separate file

### PR DESCRIPTION
**Pull request by akorosmezey**

### CODE REVIEW

File: docs/design/chalang.md [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+
+[Chalang](https://github.com/aeternity/chalang) is a stack-based language implementing smart contracts.
+
+As an example, here is the code used for lightning payments, annotated:
+
+## lightning code
+
+It is expected to return a stack with at least 4 values: [ShareRoot, Amount, Nonce, Delay | _].
+
+``stack_size int 0 ==
```
Comment (lucafavatella) [inline docs/design/chalang.md]: Using 3 backticks on its own line would have made code keep newlines.

File: docs/design/chalang.md [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+if int 0 int 2 int 10000
+else int 49 int 1 int 0
+then nil crash``
+
+*   `stack_size`
+    measures the current size of the hash and pushes it on (the top of) the stack as a number
+*   `int 0`
+    pushes the integer 0 on the stack
+*   `==`
+    compares the last to elements of the stack. pushes 1 (true) if they are equalm 0 otherwise. Compared elements are kept on the stack.
```
Comment (lucafavatella) [inline docs/design/chalang.md]: to -> two

Comment (lucafavatella) [inline docs/design/chalang.md]: equalm -> equal,

Comment (wagerlabs) [inline docs/design/chalang.md]: +1

**Comment by lucafavatella** [comment]
> Please fix the backticks.

I will prepare local branch with commit added fixing backticks fixes. (I understand @akorosmezey may not be able to address comments today.)

**Comment by lucafavatella** [comment]
Opened #338 addressing comments.

**Pull request closed without merging**